state-synchronizer/src/coordinator.rs

```diff
@@ -299,63 +316,93 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
     async fn process_chunk_request(
         &mut self,
         peer_id: PeerId,
-        mut request: GetChunkRequest,
+        request: GetChunkRequest,
     ) -> Result<()> {
-        if request.timeout > self.config.max_timeout_ms
-            || request.limit > self.config.max_chunk_limit
-        {
-            bail!(
-                "[state sync] Request timeout: {}, chunk limit: {}; configured max timeout is {} ms, and chunk limit is {}",
-                request.timeout,
-                request.limit,
-                self.config.max_timeout_ms,
-                self.config.max_chunk_limit
-            );
-        }
-        let target = request
-            .ledger_info_with_sigs
-            .take()
-            .map(TryInto::try_into)
-            .transpose()?;
         self.sync_state_with_local_storage().await?;
-        let local_li_version = self.local_state.highest_local_li.ledger_info().version();
         debug!(
-            "[state sync] chunk request: peer_id: {}, request known version: {}, target version: {}, local li version: {}",
+            "[state sync] chunk request: peer_id: {}, local li version: {}, req: {}",
             peer_id.short_str(),
-            request.known_version,
-            target.as_ref().map_or("None".to_string(), |t: &LedgerInfoWithSignatures| t.ledger_info().version().to_string()),
-            local_li_version,
+            self.local_state.highest_local_li.ledger_info().version(),
+            request,
         );
 
+        let sender = self
+            .peer_manager
+            .get_network_sender(&peer_id)
+            .ok_or_else(|| format_err!("ChunkRequest from unknown peer {}", peer_id.short_str()))?;
+        match request.target().clone() {
+            TargetType::TargetLedgerInfo(li) => {
+                self.process_request_target_li(sender, peer_id, request, li)
+                    .await
+            }
+            TargetType::HighestAvailable { timeout_ms } => {
+                self.process_request_highest_available(sender, peer_id, request, timeout_ms)
+                    .await
+            }
+            TargetType::Waypoint(_) => panic!("Waypoint processing not supported."),
```

Comment:
I'm sure this is going to be fixed before v1, but I'd sleep better if we just logged this as an error for the moment :D

Reply:
Yea, this is the skeleton for the code structure: no worries, adding the proper implementation soon.